### PR TITLE
Add config file option to init script

### PR DESCRIPTION
**Pull request by fgallese**
Solution proposed for #240

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -423,6 +423,33 @@
...
   }
+  
+  # Ensure that the correct config file is used.
+  
+  if $::osfamily == 'debian' {
+		file { '/etc/init.d/zabbix-server':
+			ensure  => present,
+			mode    => '0755',
+			require => Package["zabbix-server-${db}"],
+			replace => true,
```
Comment (juniorsysadmin) [inline manifests/server.pp]: This is true by default, so can be removed

File: manifests/server.pp [hunk 2]
```diff
@@ -423,6 +423,33 @@
...
+			content => template('zabbix/zabbix-server-debian.init.erb'),
+		}
+  } elsif $::osfamily == 'redhat' {
+		file { '/etc/init.d/zabbix-server':
+			ensure  => present,
+			mode    => '0755',
+			require => Package["zabbix-server-${db}"],
+			replace => true,
```
Comment (juniorsysadmin) [inline manifests/server.pp]: This is true by default, so can be removed

File: manifests/server.pp (hunk 1 above)
Comment (fgallese) [inline manifests/server.pp]: So, you mean to leave only the _content_ attribute so it always builds the file from the template ?

Comment (juniorsysadmin) [inline manifests/server.pp]: ```replace => true``` is the default for the File resource

File: manifests/agent.pp [hunk 3]
```diff
@@ -312,6 +312,33 @@
     require => Class['zabbix::repo'],
     tag     => 'zabbix',
   }
+  
+  # Ensure that the correct config file is used.
+  
+  if $::osfamily == 'debian' {
+	file { '/etc/init.d/zabbix-agent':
+		ensure  => present,
```
Comment (juniorsysadmin) [inline manifests/agent.pp]: It's better to make this ```file```rather than ```present```

File: manifests/agent.pp [hunk 4]
```diff
@@ -312,6 +312,33 @@
...
+		mode    => '0755',
+		require => Package["$zabbix_package_agent"],
+		replace => true,
+		content => template('zabbix/zabbix-agent-debian.init.erb'),
+	}
+  } elsif $::osfamily == 'redhat' {
+		file { '/etc/init.d/zabbix-agent':
+			ensure  => present,
```
Comment (juniorsysadmin) [inline manifests/agent.pp]: It's better to make this ```file```rather than ```present```

File: manifests/server.pp [hunk 5]
```diff
@@ -423,6 +423,33 @@
     require => Class['zabbix::repo'],
     tag     => 'zabbix',
   }
+  
+  # Ensure that the correct config file is used.
+  
+  if $::osfamily == 'debian' {
+		file { '/etc/init.d/zabbix-server':
+			ensure  => present,
```
Comment (juniorsysadmin) [inline manifests/server.pp]: It's better to make this ```file```rather than ```present```

File: manifests/server.pp [hunk 6]
```diff
@@ -423,6 +423,33 @@
...
+			mode    => '0755',
+			require => Package["zabbix-server-${db}"],
+			replace => true,
+			content => template('zabbix/zabbix-server-debian.init.erb'),
+		}
+  } elsif $::osfamily == 'redhat' {
+		file { '/etc/init.d/zabbix-server':
+			ensure  => present,
```
Comment (juniorsysadmin) [inline manifests/server.pp]: It's better to make this ```file```rather than ```present```

**Comment by juniorsysadmin** [comment]
I'm looking into why the TravisCI tests are failing. It's not because of the code, but your code will fail puppet-lint with those indents.

**Comment by bastelfreak** [comment]
Hi,

could you explain to me why you need this? managing init scripts is always very tricky. I would try to avoid it if possible. Besides the sysv scripts we also need systemd units.

**Comment by fgallese** [comment]
Well, the fact thet you allow to change the path/name of the config path but if you do so it breaks the installation seems off to me.

Personally, I've always used /etc/zabbix/server.conf and /etc/zabbix/agentd.conf, so I changed the value of these in the puppet module. That broke the installation as it didn't start the daemons cause it was looking for the /etc/zabbix/zabbix_agentd.con file.

I saw it as an easy fix, if you change the path/name of the config files I will know it, so why dont I correct the init script so that it doesnt break the installation?

That is the reason why I thought the change was an improvement.

**Comment by fgallese** [comment]
If you are intereses in keeping this change I can add the systemd files.

**Comment by bastelfreak** [comment]
Hi,
please add the systemd files as well. Please take a look at the discussion here: https://github.com/voxpupuli/puppet-kafka/pull/82#discussion-diff-69626042

we should throw modified systemd units into /etc/systemd, not /usr. 

---

Can you please check your configured git email? github is unable to recognize your commits (there isn't your avatar next to your commits).

**Comment by fgallese** [comment]
@bastelfreak I've fixed the github email missconfigurarion. I will now add the systemd as you stated in your message and make a new pull request.

**Comment by fgallese** [comment]
Hello @bastelfreak 

I've added systemd templates on my latest commit.

File: manifests/agent.pp [hunk 7]
```diff
@@ -312,6 +312,57 @@
...
   }
+  
+  # Ensure that the correct config file is used.
+  
+  if $::osfamily == 'debian' {
+    if $::operatingsystemrelease !~ /^8.*/ {
+      file { '/etc/init.d/zabbix-agent':
+        ensure  => file,
+        mode    => '0755',
+        require => Package["$zabbix_package_agent"],
```
Comment (bastelfreak) [inline manifests/agent.pp]: the quotes here aren't needed

File: manifests/agent.pp [hunk 8]
```diff
@@ -312,6 +312,57 @@
     require => Class['zabbix::repo'],
     tag     => 'zabbix',
   }
+  
+  # Ensure that the correct config file is used.
+  
+  if $::osfamily == 'debian' {
+    if $::operatingsystemrelease !~ /^8.*/ {
```
Comment (bastelfreak) [inline manifests/agent.pp]: what happens when debian 9 gets released?

File: templates/zabbix-agent-systemd.init.erb [hunk 9]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Zabbix Agent
+After=syslog.target network.target
+
+[Service]
+Type=oneshot
```
Comment (bastelfreak) [inline templates/zabbix-agent-systemd.init.erb]: this isn't a simple script, zabbix is a daemon. so oneshot is probably wrong. forking should be correct

Comment (fgallese) [inline templates/zabbix-agent-systemd.init.erb]: You are right acording to [redhat documentation](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System_Administrators_Guide/sect-Managing_Services_with_systemd-Unit_Files.html).

I'll fix it along with your other comments.

**Comment by bastelfreak** [comment]
Here are two examples for the unit files:
https://aur.archlinux.org/cgit/aur.git/tree/zabbix-agent.service?h=zabbix3
https://aur.archlinux.org/cgit/aur.git/tree/zabbix-server.service?h=zabbix3

**Comment by fgallese** [comment]
@bastelfreak Thank you. I've reviewed them and after[ checking each parameter ](https://www.freedesktop.org/software/systemd/man/systemd.service.html)I've included them in my latest commit.

**Comment by bastelfreak** [comment]
@fgallese looks good! sorry that this takes so long :(

two things, then we should be done:
the systemd units need to notify systemd. otherwise they arent recognized. I recommend https://github.com/camptocamp/puppet-systemd for this. We already use it in the kafka module.
* add it as a dependency to metadata.json
* add it to .fixtures.yml
* include the class in a sane place
* the file resources that create units should make a notify to the Exec

----

I'm currently doing a modulesync in #245, this should fix your travis issues.

**Comment by fgallese** [comment]
@bastelfreak: Thank you, I've made the changes you mentioned.

**Comment by bastelfreak** [comment]
@fgallese thanks \o/

looks good, could you now please rebase against latest master? Then the tests should turn green and I can merge your PR.

**Comment by bastelfreak** [comment]
￼I am going to close this PR since. The upstream branch is gone and we implemented it already.

**Pull request closed without merging**